lib/OnyxUtils.ts

```diff
@@ -248,78 +252,54 @@ function getAllKeys() {
         return cache.getAllKeys();
     });
 
-    return cache.captureTask(taskName, promise);
+    return cache.captureTask(taskName, promise) as Promise<Set<OnyxKey>>;
 }
 
 /**
  * Checks to see if the a subscriber's supplied key
  * is associated with a collection of keys.
- *
- * @param {String} key
- * @returns {Boolean}
  */
-function isCollectionKey(key) {
+function isCollectionKey(key: OnyxKey): key is CollectionKeyBase {
     return onyxCollectionKeyMap.has(key);
 }
 
-/**
- * @param {String} collectionKey
- * @param {String} key
- * @returns {Boolean}
- */
-function isCollectionMemberKey(collectionKey, key) {
+function isCollectionMemberKey<TCollectionKey extends CollectionKeyBase>(collectionKey: TCollectionKey, key: string): key is `${TCollectionKey}${string}` {
```

Comment:
NAB: I see that `CollectionKeyBase` is just `string`: 

<img width="431" alt="image" src="https://github.com/Expensify/react-native-onyx/assets/87341702/4bae74ea-0f0f-4a83-abc9-3b0f5d49da9b">


Is it supposed to be narrowed down later?

Reply:
It's a string in react-native-onyx but later it's narrowed down in the end repo (e.g. NewDot)

Reply:
Also playing with `ConnectOptions<TKey extends OnyxKey>` I see that it does matter if we use `CollectionKeyBase` or `Key` even if visual code just says `string`